Common subdirectories: rest-server-0.10.0-orig/changelog and rest-server-0.10.0/changelog
Common subdirectories: rest-server-0.10.0-orig/cmd and rest-server-0.10.0/cmd
Common subdirectories: rest-server-0.10.0-orig/docker and rest-server-0.10.0/docker
Common subdirectories: rest-server-0.10.0-orig/examples and rest-server-0.10.0/examples
Common subdirectories: rest-server-0.10.0-orig/.github and rest-server-0.10.0/.github
diff -u rest-server-0.10.0-orig/handlers_test.go rest-server-0.10.0/handlers_test.go
--- rest-server-0.10.0-orig/handlers_test.go	2020-09-13 11:24:29.000000000 +0200
+++ rest-server-0.10.0/handlers_test.go	2022-01-26 23:12:00.905436678 +0100
@@ -16,25 +16,24 @@
 
 func TestJoin(t *testing.T) {
 	var tests = []struct {
-		base, name string
-		result     string
-	}{
-		{"/", "foo/bar", "/foo/bar"},
-		{"/srv/server", "foo/bar", "/srv/server/foo/bar"},
-		{"/srv/server", "/foo/bar", "/srv/server/foo/bar"},
-		{"/srv/server", "foo/../bar", "/srv/server/bar"},
-		{"/srv/server", "../bar", "/srv/server/bar"},
-		{"/srv/server", "..", "/srv/server"},
-		{"/srv/server", "../..", "/srv/server"},
-		{"/srv/server", "/repo/data/", "/srv/server/repo/data"},
-		{"/srv/server", "/repo/data/../..", "/srv/server"},
-		{"/srv/server", "/repo/data/../data/../../..", "/srv/server"},
-		{"/srv/server", "/repo/data/../data/../../..", "/srv/server"},
-	}
+                base   string
+                names  []string
+                result string
+        }{
+                {"/", []string{"foo", "bar"}, "/foo/bar"},
+                {"/srv/server", []string{"foo", "bar"}, "/srv/server/foo/bar"},
+                {"/srv/server", []string{"foo", "..", "bar"}, "/srv/server/foo/bar"},
+                {"/srv/server", []string{"..", "bar"}, "/srv/server/bar"},
+                {"/srv/server", []string{".."}, "/srv/server"},
+                {"/srv/server", []string{"..", ".."}, "/srv/server"},
+                {"/srv/server", []string{"repo", "data"}, "/srv/server/repo/data"},
+                {"/srv/server", []string{"repo", "data", "..", ".."}, "/srv/server/repo/data"},
+                {"/srv/server", []string{"repo", "data", "..", "data", "..", "..", ".."}, "/srv/server/repo/data/data"},
+        }
 
 	for _, test := range tests {
 		t.Run("", func(t *testing.T) {
-			got, err := join(filepath.FromSlash(test.base), test.name)
+			got, err := join(filepath.FromSlash(test.base), test.names...)
 			if err != nil {
 				t.Fatal(err)
 			}
